plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -3187,6 +3207,12 @@ public static function shipping_zone_methods_save_changes() {
 					$option_key      = $shipping_method->get_instance_option_key();
 					if ( $wpdb->delete( "{$wpdb->prefix}woocommerce_shipping_zone_methods", array( 'instance_id' => $instance_id ) ) ) {
 						delete_option( $option_key );
+						/**
+						 * Notifies that a non-option setting has been deleted.
+						 *
+						 * @since 7.8.0
+						 */
+						do_action( 'woocommerce_delete_non_option_setting', array( 'id' => 'zone_method' ) );
```

Comment:
Similar as above, I think we should stick with the `woocommerce_update_non_option_setting` here and add an `action` for `delete`.